src/extension.ts

```diff
@@ -16,22 +25,32 @@ import makeGraph, { FactoryMap } from "./util/makeGraph";
  * use to record test cases.
  * - Creates an entrypoint for running commands {@link CommandRunner}.
  */
-export async function activate(context: vscode.ExtensionContext) {
+export async function activate(
+  context: vscode.ExtensionContext,
+): Promise<CursorlessApi> {
   const { getNodeAtLocation } = await getParseTreeApi();
   const commandServerApi = await getCommandServerApi();
 
-  const graph = makeGraph(
-    {
-      ...graphFactories,
-      extensionContext: () => context,
-      commandServerApi: () => commandServerApi,
-      getNodeAtLocation: () => getNodeAtLocation,
-    } as FactoryMap<Graph>,
-    ["ide"],
-  );
+  if (isTesting()) {
+    // FIXME: At some point we'll probably want to support partial mocking
+    // rather than mocking away everything that we can
+    const fake = new FakeIDE();
+    fake.mockAssetsRoot(context.extensionPath);
+    injectIde(fake);
```

Comment:
If we mock at this point, then we don't need to design our components to expect the `ide` singleton to change, which would be painful and break the contract of a singleton

Tempting to just not support injecting more than once, but we're currently using that functionality for spies.  Might revisit down the line